src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java

```diff
@@ -67,7 +67,8 @@ public StringProperty filterTextProperty() {
      * We need to notify the {@link StateManager} about this change so that the main table gets updated.
      */
     private void onSelectedGroupChanged(GroupNodeViewModel newValue) {
-        if (currentDatabase != stateManager.activeDatabaseProperty().get()) {
+        //TODO: Can one of this be empty?
+        if (currentDatabase.get() != stateManager.activeDatabaseProperty().get().get()) {
```

Comment:
@tobiasdiez  I am not sure if we should compare the values or if it is enough to check the object refs